packages/material-ui/src/MenuList/MenuList.js

```diff
@@ -6,66 +6,95 @@ import ReactDOM from 'react-dom';
 import warning from 'warning';
 import ownerDocument from '../utils/ownerDocument';
 import List from '../List';
+import getScrollbarSize from 'dom-helpers/util/scrollbarSize';
+
+function MenuList({ actions, children, className, onBlur, onKeyDown, disableListWrap, ...other }) {
+  const [currentTabIndex, setCurrentTabIndex] = React.useState(null);
+  const blurTimerRef = React.useRef();
+  const listRef = React.useRef();
+  const selectedItemRef = React.useRef();
```

Comment:
Without looking at how we write to this it's not obvious what this ref is holding. It doesn't add any information:

- `listRef`: a ref to `findDOMNode`
- `selectedItemRef`: a ref to `findDOMNode`
- `blurTimerRef`: a ref to a timeout id

Even more problematic is that within this change it is assumed at several lines that `selectedItemRef` is a ref to a component and `findDOMNode` is called unnecessarily.

We  should really enforce "`*Ref` only for variables that are passed to component `ref` or `*Ref` props". 

Since we assign a DOM node to it we should go with 
```suggestion
  const selectedItem = React.useRef();
```
To the argument "it's not obvious what the shape is": Yes it is. You only have to look at the declaration and the name. With the generic `*Ref` you only have to look at the name but the content is misleading since most people are still in the mindset of `*Ref` is component ref.

/cc @oliviertassinari @mbrookes

Reply:
> We should really enforce "*Ref only for variables that are passed to component ref or *Ref props".

I'm happy with this strategy.

Reply:
What would be your preference for cases like the following?
```
const list = listRef.current;
```

If I change `listRef` to be just `list`, then I'm not sure what naming standard to use for the `list` variable above. One option is for the variable to not exist and just use `list.current` in its place, but in functions like `handleKeyDown` where `list` is referred to a lot of places I think the code readability will suffer a little to use `list.current` in all those places.

Reply:
Controversial but what does everyone think about 
```
const { current: list } = React.useRef();
```

Reply:
That wouldn't support setting `list` in any way that will be retained across renders.
```
const { current: list } = React.useRef();
// This won't change the ref's `current` property,
// it just changes what the local `list` variable points at.
list = ReactDOM.findDOMNode(ref);
```

Reply:
@eps1lon In the flurry of comments about `findDOMNode`, you might have missed my comment [here](https://github.com/mui-org/material-ui/pull/14865#discussion_r265137881). If we don't end these refs with "Ref", my main naming uncertainty is when we set a variable to the ".current" of one of those refs. I find it very clear to use:
```
const x = xRef.current;
```
The main alternatives I can think of are:

- avoid creating these variables -- just use `x.current` everywhere even if rather verbose
- use a different suffix (e.g. "Ptr" for pointer: `const x = xPtr.current`) for refs that aren't traditional refs

I don't have strong feelings about which is better. The structure of my code using the more concise local variable in `handleKeyDown`:
```
const list = listRef.current;
```
is a direct translation of the code from the class-based version:
```
const list = this.listRef;
```

Reply:
My main concern was that people falsely assume that `*Ref` is in relation to host components. This was confirmed with the redundant `findDOMNode(selectedItemRef.current)` call when before you did `ref => (selectedItemRef.current = React.findDOMNode(ref))`.

We can keep on suffixing any return value from `useRef` or `createRef` but I fear that this will not push people to a different mental model about these things. `useRef` is not just the hook form of `createRef`.

Maybe this was just an outlier and as long as we're able to catch these in review we might as well stick with the suffixing.

Reply:
The redundant `findDOMNode` call was just a direct carry over of the existing [`Menu`](https://github.com/mui-org/material-ui/blob/next/packages/material-ui/src/Menu/Menu.js#L43) code which was already redundant. I almost removed it, but decided to leave it to keep the logic exactly equivalent to the pre-hook version.

Reply:
I ran into both these concerns (`const x = xRef.current;` carried over from `const x = this.xRef;` and redundant findDOMNode, which I removed) in some the components I converted.

I'm happy to match whatever pattern is agreed here for the the `xRef` question.